sdk/search/Azure.Search.Documents/src/AzureSearchDocumentsEventSource.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.Tracing;
+using Azure.Core.Diagnostics;
+
+namespace Azure.Search.Documents
+{
+    [EventSource(Name = EventSourceName)]
+    internal sealed class AzureSearchDocumentsEventSource : EventSource
+    {
+        private const string EventSourceName = "Azure-Search-Documents";
+
+        internal const int BatchActionCountUpdatedEvent = 1;
+        internal const int BatchSubmittedEvent = 2;
+        internal const int RetryQueueResizedEvent = 3;
+        internal const int PendingQueueResizedEvent = 4;
```

Comment:
I think we're missing the most important events.  We should have something warning about when we split a batch because that means you sized things incorrectly, we should have something warning about destroying with unsubmitted work, and we should have something warning about event handlers throwing an exception we're effectively swallowing.

Reply:
I think an event logging for AutoFlush is a good idea too so people know how long they leave items sitting.